RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractZoneIdentifier.java

```diff
@@ -0,0 +1,398 @@
+            }
+        }
+    }
+
+    /**
+     * Find a list of all the file that end in <i>:Zone.Identifier<i>.
+     *
+     * @param dataSource Content
+     *
+     * @return A list of zone identifier files
+     *
+     * @throws TskCoreException
+     */
+    private List<AbstractFile> findZoneFiles(Content dataSource) throws TskCoreException {
+        org.sleuthkit.autopsy.casemodule.services.FileManager fileManager
+                = currentCase.getServices().getFileManager();
+
+        return fileManager.findFiles(dataSource, ZONE_IDENIFIER_FILE);
+    }
+
+    /**
+     * Find the file that the Zone.Identifer file was created alongside.
+     *
+     * @param dataSource Content
+     * @param zoneFile   The zone identifier case file
+     *
+     * @return The downloaded file or null if a file was not found
+     *
+     * @throws TskCoreException
+     */
+    private AbstractFile getDownloadFile(Content dataSource, AbstractFile zoneFile) throws TskCoreException {
+        AbstractFile downloadFile = null;
+
+        org.sleuthkit.autopsy.casemodule.services.FileManager fileManager
+                = currentCase.getServices().getFileManager();
+
+        String downloadFileName = zoneFile.getName().replace(ZONE_IDENIFIER, ""); //NON-NLS
+
+        List<AbstractFile> fileList = fileManager.findFiles(dataSource, downloadFileName, zoneFile.getParentPath());
+
+        if (fileList.size() == 1) {
+            downloadFile = fileList.get(0);
+
+            // Check that the download file and the zone file came from the same dir
+            if (!downloadFile.getParentPath().equals(zoneFile.getParentPath())) {
+                downloadFile = null;
+            } else if (zoneFile.getMetaAddr() != downloadFile.getMetaAddr()) {
+                downloadFile = null;
+            }
+        }
+
+        return downloadFile;
+    }
+
+    /**
+     * Create a Download Source Artifact for the given ZoneIdentifierInfo
+     * object.
+     *
+     * @param downloadFile AbstractFile representing the file downloaded, not
+     *                     the zone indentifier file.
+     * @param zoneInfo     Zone Indentifer file wrapper object
+     *
+     * @return TSK_DOWNLOAD_SOURCE object for given parameters
+     */
+    private BlackboardArtifact createDownloadSourceArtifact(AbstractFile downloadFile, ZoneIdentifierInfo zoneInfo) {
+
+        Collection<BlackboardAttribute> bbattributes = new ArrayList<>();
+
+        bbattributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_URL,
+                RecentActivityExtracterModuleFactory.getModuleName(),
+                (zoneInfo.getURL() != null) ? zoneInfo.getURL() : "")); //NON-NLS
+
+        bbattributes.add(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DOMAIN,
+                RecentActivityExtracterModuleFactory.getModuleName(),
+                (zoneInfo.getURL() != null) ? NetworkUtils.extractDomain(zoneInfo.getURL()) : "")); //NON-NLS
```

Comment:
apache StringUtils.defaultString() does exactly this coersion of null to an empty string, and I find it easier to read.